server/executor/assertion_runner.go

```diff
@@ -5,42 +5,43 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/kubeshop/tracetest/server/assertions"
 	"github.com/kubeshop/tracetest/server/model"
 )
 
 type AssertionRequest struct {
+	Ctx  context.Context
```

Comment:
We shouldn't use ctx as a property. Instead, we should always pass it as a func param (https://go.dev/blog/context-and-structs). You could instead do something like:

```
import "go.opentelemetry.io/otel/propagation"

type AssertionRequest struct {
  carrier propagation.TextMapCarrier
  //...
}
```

then, in line 66:
```
ctx := otel.GetTextMapPropagator().Extract(context.Background(), assertionRequest.carrier)
err := e.runAssertionsAndUpdateResult(ctx, assertionRequest)
```